pykit/Toolkit.py

```diff
@@ -0,0 +1,24 @@
+class Toolkit:
+    """
+    Python Toolkit for interacting with the IBM i via different transports.
+    """
+    def __init__(self, connection):
+        self.connection = connection
+        self.payload = ''
+
+    def add(self, o):
+        """
+        Add object to the transport's payload.
+
+        :param o: Object to be added to the transport
+        :return: void
+        """
+        self.connection.add(o)
```

Comment:
Why does the connection store the options? It seems that the toolkit is merely a wrapper around the connection, in which case you might as well just get rid of it.

My idea with the separation of toolkit object and connection/transport object is that the toolkit object gathers up actions to be executed and then when you execute it, it converts the actions in to a payload which it passes to the connection/transport object. Basic separation of concerns: a toolkit object knows how to convert actions to JSON and a connection knows how to pass that JSON to db2sock and back.

Reply:
Now that I'm getting into this finally, I'm wondering what would be the best way to go about this. I had already noticed that Toolkit looked like a wrapper at the moment, but that's because I have it so generic right now, and there's only one transport.

My original thought was that the `add`method in Toolkit is supposed to translate the object and then add it to the connection, but right now the connection only accepts "raw json". I see now that is probably unnecessary.

So you're saying you don't want to add anything to the connection ever. Only use the connection during execution?

Reply:
You probably don't want to make the connection method public, but then the toolkit method can't call it, unless you get tricky:

```python
class Conn:
    def __execute(self, payload):
        # pass JSON payload to db2sock and return the result
    def toolkit(self):
        tk = Toolkit(self, self.__execute)

class Toolkit:
    def __init__(conn, conn_exec):
        self.conn = conn
        self.conn_exec = conn_exec

    def execute(self):
        payload = {} # convert actions to json
        self.conn_exec(self.conn, payload)
```

Reply:
Sounds good. We're on the same page now, then.